### PR TITLE
Download binary from REPO instead of building

### LINKED ISSUE #122: Support chain binary download
Add an option to download the binary instead of building it from source.

### PR DESCRIPTION
**Pull request by YurkoWasHere**
PR will allow downloading pre-compiled binaries from the source chain's repo instead of compiling them.
Closes #122 

New variables added:

### gaiad_binary_source
**values:**
`build` (default) will compile a gaiad binary as it did before
`release` will download the gaiad binary from the git repo based on `gaiad_version` and `gaiad_binary_release`

### gaiad_binary_release
url to release binary. Uses `gaiad_version` to target bin
**value:** string

### CODE REVIEW

**Review by dasanchez — CHANGES_REQUESTED** [review]
Tested with `examples/inventory-local.yml`:
✅ without modifications
✅ `use_cosmovisor: false`
✅ `gaiad_binary_source: release`
✅ `use_cosmovisor: false` and `gaiad_binary_source: release`
Thanks @YurkoWasHere !

File: roles/gaia/tasks/main.yml [hunk 1]
```diff
@@ -114,8 +114,28 @@
   register: gaiad_current_version
   ignore_errors: true
 
+## Creatre go bin directory
```
[the remark this answers is not in the document]
Comment (YurkoWasHere) [reply roles/gaia/tasks/main.yml:117]:
```suggestion
## Create go bin directory
```

File: roles/gaia/tasks/main.yml [hunk 2]
```diff
@@ -114,8 +114,28 @@
   register: gaiad_current_version
   ignore_errors: true
 
+## Create go bin directory
+- name: Prepare cosmovisor folder
```
Comment (YurkoWasHere) [inline roles/gaia/tasks/main.yml:118]:
```suggestion
- name: Prepare go folder
```

File: roles/gaia/tasks/main.yml [hunk 3]
```diff
@@ -114,8 +114,28 @@
   register: gaiad_current_version
   ignore_errors: true
 
+## Creatre go bin directory
+- name: Prepare cosmovisor folder
+  when: gaiad_binary_source == "release"
+  file:
+    path: '{{ gaiad_user_home }}/go/bin'
```
[the remark this answers is not in the document]
Comment (YurkoWasHere) [reply roles/gaia/tasks/main.yml:121]: Thanks, 
Copy paste fail

**Review by dasanchez — APPROVED** [review]

**Pull request merged**